External libraries refactoring

XavierCLL opened the pull request:
- Refactoring the process of generating (paver) and importing (init) the external libraries required by the plugin because it is not strictly OS-dependent: extlibs_OS > extlibs
- Now is enough to have a Github action for only one OS
- Fixes some paver packaging

With this, we are going to reduce the complexity of generating and packaging the external libraries (and package size). We need to check the new extlibs through two tests ( @gena and @SiggyF ? )

| Test | Windows | Linux | Mac |
| ---- | --------- | -------- | ---- |
| GEE Authentication | :white_check_mark: Xavier <br/> tester 2 | :white_check_mark: Xavier <br/> tester 2   | :white_check_mark: SiggyF <br/> tester 2 |
| GEE Example | :white_check_mark: Xavier <br/> tester 2 | :white_check_mark: Xavier <br/> tester 2  |:white_check_mark: SiggyF<br/> tester 2 |

For testing, use the Artifact at https://github.com/gee-community/qgis-earthengine-plugin/actions/runs/8509882708 (the package is created by Ubuntu OS but should work for any OS)

SiggyF commented on requirements.txt:
```diff
@@ -1,5 +1,4 @@
 ratelim
-pyCrypto
-earthengine-api
+earthengine-api==0.1.335
```
```suggestion
earthengine-api
```
I would not pin this version. Or did you mean `>=`?

XavierCLL replied:
Fixed thanks

SiggyF commented on pavement.py:
```diff
@@ -3,41 +3,49 @@
 )
 
 
+def clean_extlibs():
```
Can we do without this? I find it a bit scary to do an `os.walk` + `os.remove` on users computers. 
Or perhaps remove the specific directory?

XavierCLL replied:
After some testing, I realized that the dependencies of this plugin are not OS-depend, you don't want to have windows DLLs in your MacOS plugin dir and so on. So this deletes any unneeded linked lib and caches. Beside:

- This function runs only in the ext_lib directory created when pave is call
- This function is part of the plugin packaging tool that runs in a docker container inside Github action, in theory a user never needs to run this
- Another purpose of this is to reduce the plugin size

SiggyF reviewed (commented):
The file is now double zipped with linux in the name:
name: ee_plugin-${{ matrix.os }}.zip

SiggyF commented:
I think the pycrypto library was the only native (c-based) dependency (which also required manual compilation). If that's no longer needed, we can see the distribution as an "all" / "any" / "python only" distribution. 

I left a few minor comments:
- crawling and removing files feels a bit uncomfortable. 
- the earthengine version is now pinned, I added a comment to revert that.
- It's named linux, while it's now all/any.  The package is now double zipped.

You could also consider keeping it as is. Once there will be a system specific dependency added back, we need to revert back to this approach.  

I tested it, and it works on my m1 mac.

XavierCLL commented:
> The file is now double zipped with linux in the name: name: ee_plugin-${{ matrix.os }}.zip

That is a Github action issue/limitation

XavierCLL commented:
> I think the pycrypto library was the only native (c-based) dependency (which also required manual compilation). If that's no longer needed, we can see the distribution as an "all" / "any" / "python only" distribution.
> 
> I left a few minor comments:
> 
>     * crawling and removing files feels a bit uncomfortable.
> 
>     * the earthengine version is now pinned, I added a comment to revert that.
> 
>     * It's named linux, while it's now all/any.  The package is now (probably was already) double zipped.
> 
> 
> You could also consider keeping it as is. Once there will be a system specific dependency added back, we need to revert back to this approach.
> 
> I tested it, and it works on my m1 mac.

Hi @SiggyF thanks so much for your review!

As soon as I realized that EE API no longer needed pycrypto, I tested and created this PR to have an independent OS for the external libs for the plugin. And, of course, if in the future the EE-API includes a library, not OS-independent, we can roll back to the old code. The good thing to have this, simplified and not OS-depend: is easy to maintain and release it, a lot of size reduction for the final plugin package, fewer GitHub actions.

Regards!
Xavier

gena commented on .github/workflows/qgis-plugin-package.yml:
```diff
@@ -16,8 +16,8 @@ jobs:
       matrix:
-        python-version: ["3.7"]
-        os: [macos-latest, windows-latest, ubuntu-latest]
+        python-version: ["3.10"]
```
Please upgrade the https://github.com/gee-community/qgis-earthengine-plugin/blob/master/dependencies/linux/Dockerfile to 3.10 as well to keep Python versions consistent

gena commented on requirements.txt:
```diff
@@ -1,5 +1,4 @@
 ratelim
-pyCrypto
-earthengine-api
+earthengine-api>=0.1.335
```
Please change this to a specific version to avoid mismatch between EE versions for users, which will be harder to debug in case of bugs. For example, 1.20 would be good here.

XavierCLL replied:
I think that it is not necessary due that those libs are distributed inside the final package (ran by the github action inside the docker), so users never install it directly. But, before the next release we need to update the "extlibs" and set up the latest versions of those libs
